From 7b78154ef194ab672336a447c8381b9eed279c6c Mon Sep 17 00:00:00 2001
From: Pranav Vashi <neobuddy89@gmail.com>
Date: Thu, 08 Oct 2026 18:43:49 +0530
Subject: [PATCH] Launcher3: Remove glitchy IME switcher button on SUW screen

* IME switcher contextual button appears just next to back button when typing.
* Since there is no other IME installed, this button makes no sense.
* Let's remove contextual button for clean look.

Change-Id: I1cb56d9eff218289509caeca37280d66c599eefc
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---

diff --git a/quickstep/src/com/android/launcher3/taskbar/navbutton/SetupNavLayoutter.kt b/quickstep/src/com/android/launcher3/taskbar/navbutton/SetupNavLayoutter.kt
index 5748ee0..fec0974 100644
--- a/quickstep/src/com/android/launcher3/taskbar/navbutton/SetupNavLayoutter.kt
+++ b/quickstep/src/com/android/launcher3/taskbar/navbutton/SetupNavLayoutter.kt
@@ -95,13 +95,13 @@
         mNearestTouchFrame.layoutParams = navButtonsOverallViewGroupLayoutParams
         navButtonContainer.layoutParams = navButtonsLayoutParams
 
+        endContextualContainer.removeAllViews()
+        startContextualContainer.removeAllViews()
+
         if (expressiveThemeEnabled && !context.isSimpleViewEnabled) {
             return
         }
 
-        endContextualContainer.removeAllViews()
-        startContextualContainer.removeAllViews()
-
         val contextualButtonWidth =
             resources.getDimensionPixelSize(R.dimen.taskbar_contextual_button_suw_width)
         val contextualMargin =
